test/probes/zlib.test.js

```diff
@@ -138,128 +150,157 @@ describe('probes.zlib', function () {
     })
   })
 
+  // Brotli
+  before(function (done) {
+    inputs.BrotliCompress = test;
+    outputs.BrotliDecompress = test;
+    zlib.brotliCompress(test, function (err, res) {
+      if (err) return done(err);
+      inputs.BrotliDecompress = res;
+      outputs.BrotliCompress = res;
+      done();
+    })
+  })
+
   //
   // Tests
   //
   describe('async', function () {
     methods.forEach(function (method) {
       const className = upperFirst(method)
-      if (zlib[method]) {
-        it('should support ' + method, function (done) {
-          helper.test(emitter, function (done) {
-            zlib[method](inputs[className], function (err, buf) {
-              if (err) return done(err)
-              buf.toString().should.equal(outputs[className].toString())
-              done()
-            })
-          }, [
-            function (msg) {
-              checks.entry(msg)
-              msg.should.have.property('Operation', method)
-              msg.should.have.property('Async', true)
-            },
-            function (msg) {
-              checks.exit(msg)
-            }
-          ], done)
-        })
-      }
+
+      it(`should support ${method}`, function (done) {
+        // make sure the function exists
+        expect(zlib).property(method);
+        expect(typeof zlib[method]).equal('function');
```

Comment:
Not sure this should be an assertion in a test. 
I think the instrumentation implementation should check if the function exists and tests should test the behavior if the function does not exist.

Reply:
it originally just didn't do the test if the function was missing; that was clearly wrong as missing functions wouldn't be noticed at all. so rather than not testing i changed it to fail if the method was missing.

the instrumentation does check to verify that the function exists.

so the question is whether the test verifies that the function is present and then tests it or whether it just tests it. something is wrong if standard methods are not present. it seems like that's higher order than trying to run the test if they're missing; it's a more specific diagnostic of the problem.

what do you think?

Reply:
yes, I thinks it is fine, I now understand the practicality of it. 
I'm still wondering what the behavior of the instrumentation is, when the function does not exist.

Reply:
the function would just be ignored if it didn't exist.

see the wrap function (lines 83, 111) where a check is made to see if the method exists. if it doesn't exist then nothing is patched (because there is nothing to patch). a logMissing() entry is made (that's 'patching' level with the prefix of the module name, 'zlib' in this case.